src/workspacer.Shared/Layout/DwindleLayoutEngine.cs

```diff
@@ -0,0 +1,125 @@
+        }
+
+        public string Name => "dwindle";
+
+        public IEnumerable<IWindowLocation> CalcLayout(IEnumerable<IWindow> windows, int spaceWidth, int spaceHeight)
+        {
+            var list = new List<IWindowLocation>();
+            var numWindows = windows.Count();
+
+            if (numWindows == 0)
+            {
+                return list;
+            }
+
+            int numInPrimary = Math.Min(GetNumInPrimary(), numWindows);
+
+            int primaryWidth = (int)(spaceWidth * (_primaryPercent + _primaryPercentOffset));
+            int primaryHeight = spaceHeight / numInPrimary;
+            int height = spaceHeight / Math.Max(numWindows - numInPrimary, 1);
+
+            if (numInPrimary >= numWindows)
+            {
+                primaryWidth = spaceWidth;
+            }
+
+            bool curOrientation = Orientation.Vertical;
+            int curWidth = secondaryWidth;
+            int curTop = 0;
+            int curLeft = primaryWidth;
+            int curHeight = numWindows > 2 ? spaceHeight / 2 : spaceHeight;
+
+            for (var i = 0; i < numWindows; i++)
+            {
+                if (i < numInPrimary)
+                {
+                    list.Add(new WindowLocation(0, i * primaryHeight, primaryWidth, primaryHeight, WindowState.Normal));
+                }
+                else
+                {
+                    list.Add(new WindowLocation(curLeft, curTop, curWidth, WindowState.Normal));
+                    if (curOrientation == Orientation.Vertical)
+                    {
+                        curTop += curHeight;
+                        if (i < numWindows - 2)
+                        {
+                            curWidth /= 2;
+                        }
+                        curOrientation = Orientation.Horizontal;
+                    }
+                    else
+                    {
+                        curLeft += curWidth;
+                        if (i < numWindows - 2)
+                        {
+                            curHeight /= 2;
+                        }
+                        curOrientation = Orientation.Vertical;
+                    }
+                }
+            }
+            return list;
+        }
+
+        public void ShrinkPrimaryArea() {
+            _primaryPercentOffset -= _primaryPercentIncrement;
+        }
+
+        public void ExpandPrimaryArea() {
+            _primaryPercentOffset += primaryPercentIncrement;
+        }
+
+        public void ResetPrimaryArea()
+        {
+            _primaryPercentOffset = 0;
+        }
+
+        public void IncrementNumInPrimary()
+        {
+            _numInPrimaryOffset++
+        }
+
+        public void DecrementNumInPrimary()
+        {
+            if (GetNumInPrimary() > 1)
+            {
+                _numInPrimaryOffset--;
+            }
+        }
+
+        private int GetNumInPrimary()
+        {
+            return _numInPrimary + _numInPrimaryOffset;
+        }
+    }
+}
```

Comment:
This may be a solid new layout for us to consider, but I don't think they belongs in the same PR as one for changing presentation of battery-levels.

Each new feature should IMO (and by Github conventions) have its own PR, so one can independently discuss the new feature and the code implementing it, not to mention merge it on a individual level, per feature.

Please submit layouts as a separate PR, and then that can be discussed there :smiley:

Reply:
I'll open up another PR for layouts